nrf-softdevice/src/ble/connection.rs

```diff
@@ -543,6 +562,44 @@ impl Connection {
         ConnectionIter(0)
     }
 
+    /// Initiate a Data Length Update procedure.
+    ///
+    /// Note that this just initiates the data length update, it does not wait for completion.
+    /// Immediately after return, the active data length will still be the old one, and after some time they
+    /// should change to the new ones.
+    #[cfg(any(feature = "s113", feature = "s132", feature = "s140"))]
+    pub fn data_length_update(
+        &mut self,
+        params: Option<&raw::ble_gap_data_length_params_t>,
+    ) -> Result<raw::ble_gap_data_length_limitation_t, DataLengthUpdateError> {
+        let conn_handle = self.with_state(|state| state.check_connected())?;
+
+        let params = params.map(core::ptr::from_ref).unwrap_or(core::ptr::null());
+        let mut dl_limitation = unsafe { core::mem::zeroed() };
+        let ret = unsafe { raw::sd_ble_gap_data_length_update(conn_handle, params, &mut dl_limitation) };
+
+        if let Err(err) = RawError::convert(ret) {
+            warn!("sd_ble_gap_data_length_update err {:?}", err);
+            return Err(err.into());
+        }
+
+        if dl_limitation.tx_payload_limited_octets != 0 || dl_limitation.rx_payload_limited_octets != 0 {
```

Comment:
`dl_limitation` is valid on error, but you're checking it and returning it on success here.

Reply:
Oops, good catch!